protocols/request-response/Cargo.toml

```
@@ -16,12 +16,21 @@ futures = "0.3.28"
 instant = "0.1.11"
 libp2p-core = { workspace = true }
 libp2p-swarm = { workspace = true }
+libp2p-swarm-derive = { workspace = true }
 libp2p-identity = { workspace = true }
 rand = "0.8"
+serde = { version = "1.0", optional = true, features = ["derive"] }
+serde_json = { version = "1.0.96", optional = true }
+serde_cbor = { version = "0.11.2", optional = true }
 smallvec = "1.6.1"
 void = "1.0.2"
 log = "0.4.17"
 
+[features]
+default = ["json"]
+json = ["dep:serde", "dep:serde_json"]
+cbor = ["dep:serde", "dep:serde_cbor"]
```

```suggestion
json = ["dep:serde", "dep:serde_json", "libp2p-swarm/macros"]
cbor = ["dep:serde", "dep:serde_cbor", "libp2p-swarm/macros"]
```

This will give you the derive.